### ISSUE TITLE
Equal number of iteration and warmup iterations makes R hang when plotting the results.

### ISSUE THREAD

**Issue by rasmusab** [stan-dev/rstan#8]
It seems that plotting the result of a STAN model where the number of warmup iterations equaled the number of iteration causes R to hang, that is, it is not possible to stop the execution by using Ctrl-C. Looking at the documentation it is not clear to me if the `warmup` argument to `stan` specifies the number of warmup iterations *in addition* to the regular number of iteration or the number of the `iter` iterations that should be treated as warmup. If it is the latter then this might explain why R hangs. It would however be nicer if R did not hang but rather that plotting the result of the STAN model resulted in an error message.

The following code replicates the error on my computer: 

``` s
library(rstan)
model_string <- "
data {
int<lower=0> N;
real y[N];
}

parameters {
real mu;
real<lower=0> sigma;
}
model{
y ~ normal(mu, sigma);
}"

mcmc_samples <- stan(model_code=model_string, data=list(N=10, y=rnorm(10)), pars=c("mu", "sigma"), chains=3, iter=10000, warmup=10000)
plot(mcmc_samples)
```

Info regarding my current R setup: 

``` s
> sessionInfo()
R version 3.0.1 (2013-05-16)
Platform: i686-pc-linux-gnu (32-bit)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=C                 LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    graphics  grDevices utils     datasets  stats     methods  
[8] base     

other attached packages:
[1] rstan_1.3.0     inline_0.3.11   Rcpp_0.10.3     reshape_0.8.4  
[5] plyr_1.8        stringr_0.6.2   lattice_0.20-15 ggplot2_0.9.3.1

loaded via a namespace (and not attached):
 [1] codetools_0.2-8    colorspace_1.2-2   dichromat_2.0-0    digest_0.6.3      
 [5] grid_3.0.1         gtable_0.1.2       labeling_0.1       MASS_7.3-26       
 [9] munsell_0.4        proto_0.3-10       RColorBrewer_1.0-5 reshape2_1.2.2    
[13] scales_0.2.3       tools_3.0.1  
```

**Comment by rasmusab** [comment]
Hi! It seems like this issues has been fixed in Stan 2.6, R does not hang when I run my example code above. So this issues could perhaps be closed now?

**Comment by bob-carpenter** [comment]
Thanks for the heads up.

We'd strongly suggest Stan 2.7.  The R interface has a ton of improvements,
including built-in parallelism, and there's also much better error handling
at parse time and at run time.
- Bob
